pkg/orchestrator/orchestrator.go

```diff
@@ -1528,6 +1701,31 @@ func (s *ControllerServer) DeleteValue(c context.Context, req *paragliderpb.Dele
 	return &paragliderpb.DeleteValueResponse{}, nil
 }
 
+func (s *ControllerServer) GetUriFromIp(c context.Context, req *paragliderpb.GetUriFromIpRequest) (*paragliderpb.GetUriFromIpResponse, error) {
```

Comment:
This is very inefficient, especially for large numbers of tags. If we want the ability to query like this, we need to change the underlying structure of the tag service. As mentioned above, I think you should query by tag instead.